Lib/fontTools/varLib/featureVars.py

```diff
@@ -136,12 +137,14 @@ def overlayFeatureVariations(conditionalSubstitutions):
                     remainder = hashdict(remainder)
                     newMap[remainder] = newMap.get(remainder, 0) | rank
         boxMap = newMap
-    del boxMap[hashdict()]
 
     # Generate output
     items = []
     for box,rank in sorted(boxMap.items(),
                            key=(lambda BoxAndRank: -popCount(BoxAndRank[1]))):
+        # Skip the initial entire-space box only if no layers apply.
```

Comment:
The comment is not accurate.  The empty entire-space might have been broken into smaller pieces.

During the overlay there's a requirement that the boxes must cover the entire space.  During output-generation, we skip any box that doesn't have any substitution.  That's what we are doing here.

The rest looks good.  Thanks!

Reply:
So the comment should be "Skip any box that doesn't have any substitution"? :o

Reply:
> During the overlay there's a requirement that the boxes must cover the entire space. During output-generation, we skip any box that doesn't have any substitution. That's what we are doing here.

Okay I debugged that, and the way overlayBoxes works currently that can't happen.  But the change is correct and good regardless.  Adjust the comment and we are good.